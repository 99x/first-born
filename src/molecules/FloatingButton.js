import React, { Component } from "react";
import { View, StyleSheet, Platform, Dimensions } from "react-native";
import PropTypes from "prop-types";
import { Button } from "../atoms/Button";
import Icon from "react-native-vector-icons/Ionicons";

const { width, height } = Dimensions.get("window");

const color_overlay = "rgba(33, 33, 33, 0.8)";

export class FloatingButton extends Component {

    constructor(props) {
        super(props);
        this.state = {
            active: false
        }
    }

    toggleActive = () => {
        this.setState({ active: !this.state.active })
    }

    render() {
        const { children, iconName, onPress, color, ...otherProps } = this.props;

        const onButtonPress = onPress ? onPress : this.toggleActive;

        const buttonColor = color ? color : "#000"

        if (Platform.OS === "android") {
            return (
                <View style={this.state.active ? [styles.container, { backgroundColor: color_overlay }] : styles.container}>
                    {children && this.state.active && this.renderChildren(children)}
                    <View style={styles.floatingButtonAndroid}>
                        <Button rounded size={"large"} onPress={onButtonPress} color={this.state.active ? "red" : buttonColor} {...otherProps}>
                            <Icon name={iconName ? "md-" + iconName : !this.state.active ? "md-add" : "md-close"} size={22} />
                        </Button>
                    </View>
                </View>
            )
        }
        return (
            <View style={this.state.active ? [styles.container, { backgroundColor: color_overlay }] : styles.container}>
                {children && this.state.active && this.renderChildren(children)}
                <View style={styles.floatingButtonIos} >
                    <Button rounded size={"large"} onPress={onButtonPress} color={this.state.active ? "red" : buttonColor} {...otherProps}>
                        <Icon name={iconName ? "ios-" + iconName : !this.state.active ? "md-add" : "md-remove"} size={22} />
                    </Button>
                </View>
            </View>
        )
    }

    renderChildren(children) {
        if (Platform.OS === "android") {
            return (
                <View style={styles.floatingColumnAndroid} >
                    {children}
                </View>
            )
        }
        return (
            <View style={styles.floatingRowIos} >
                {children}
            </View>
        )
    }
}

FloatingButton.propTypes = {
    iconName: PropTypes.string,
    color: PropTypes.string,
    ...Button.propTypes
}

const styles = StyleSheet.create({
    container: {
        position: "absolute",
        width,
        height
    },
    floatingButtonAndroid: {
        position: "absolute",
        bottom: 30,
        right: 20
    },
    floatingButtonIos: {
        position: "absolute",
        bottom: 30,
        right: (width / 2) - 35
    },
    floatingColumnAndroid: {
        flexDirection: 'column',
        justifyContent: "space-between",
        position: 'absolute',
        flex: 1,
        bottom: 100,
        right: 20,
        alignItems: "center"
    },
    floatingRowIos: {
        flexDirection: 'row',
        justifyContent: "center",
        alignItems: "center",
        position: 'absolute',
        flex: 1,
        bottom: 90,
        paddingHorizontal: 20,
        width: "100%"
    }
})
