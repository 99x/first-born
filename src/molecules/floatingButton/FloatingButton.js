import React, { Component } from 'react'; // eslint-disable-line
import PropTypes from 'prop-types';
import { StyleSheet, Animated, Dimensions, TouchableOpacity, LayoutAnimation, Platform, Keyboard } from 'react-native';

import Icon from "react-native-vector-icons/Ionicons";

import FloatingButtonItem from './FloatingButtonItem';

import { isIphoneX } from '../../utils/platform';
import { getTouchableComponent, getRippleProps } from '../../utils/touchable';

const DEVICE_WIDTH = Dimensions.get('window').width;
const ACTION_BUTTON_SIZE = 56;

export class FloatingButton extends Component {
  constructor(props) {
    super(props);

    this.state = {
      active: false,
      keyboardHeight: 0
    };

    this.mainBottomAnimation = new Animated.Value(props.distanceToEdge + props.mainVerticalDistance);
    this.actionsBottomAnimation = new Animated.Value(ACTION_BUTTON_SIZE + props.distanceToEdge + props.actionsPaddingTopBottom + props.mainVerticalDistance);
    this.animation = new Animated.Value(0);
    this.actionsAnimation = new Animated.Value(0);
    this.visibleAnimation = new Animated.Value(props.visible ? 0 : 1);
    /*
     * this animation will fix an error on ReactNative (Android) where
     * interpolations with 0 and 1 don't work as expected.
     */
    this.fadeAnimation = new Animated.Value(props.visible ? 1 : 0);
  }

  componentDidMount() {
    const { openOnMount, listenKeyboard } = this.props;

    if (openOnMount) {
      this.animateButton();
    }

    if (listenKeyboard) {
      const showEvent = Platform.OS === 'ios' ? 'keyboardWillShow' : 'keyboardDidShow';
      const hideEvent = Platform.OS === 'ios' ? 'keyboardWillHide' : 'keyboardDidHide';
      this.keyboardWillShowListener = Keyboard.addListener(showEvent, this.onKeyboardShow);
      this.keyboardWillHideListener = Keyboard.addListener(hideEvent, this.onKeyboardHideHide);
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.visible !== this.props.visible) {
      if (nextProps.visible) {
        Animated.parallel([
          Animated.spring(this.visibleAnimation, { toValue: 0 }),
          Animated.spring(this.fadeAnimation, { toValue: 1 })
        ]).start();
      } if (!nextProps.visible) {
        Animated.parallel([
          Animated.spring(this.visibleAnimation, { toValue: 1 }),
          Animated.spring(this.fadeAnimation, { toValue: 0 })
        ]).start();
      }
    }
  }

  componentWillUnmount() {
    const { listenKeyboard } = this.props;

    if (listenKeyboard) {
      this.keyboardWillShowListener.remove();
      this.keyboardWillHideListener.remove();
    }
  }

  onKeyboardShow = (e) => {
    const { distanceToEdge, actionsPaddingTopBottom } = this.props;
    const { height } = e.endCoordinates;

    Animated.parallel([
      Animated.spring(
        this.actionsBottomAnimation,
        {
          bounciness: 0,
          toValue: (ACTION_BUTTON_SIZE + distanceToEdge + actionsPaddingTopBottom + height) - (isIphoneX() ? 40 : 0),
          duration: 250
        }
      ),
      Animated.spring(
        this.mainBottomAnimation,
        {
          bounciness: 0,
          toValue: (distanceToEdge + height) - (isIphoneX() ? 40 : 0),
          duration: 250
        }
      )
    ]).start();
  };

  onKeyboardHideHide = () => {
    const { distanceToEdge, actionsPaddingTopBottom } = this.props;

    Animated.parallel([
      Animated.spring(
        this.actionsBottomAnimation,
        {
          bounciness: 0,
          toValue: ACTION_BUTTON_SIZE + distanceToEdge + actionsPaddingTopBottom,
          duration: 250
        }
      ),
      Animated.spring(
        this.mainBottomAnimation,
        {
          bounciness: 0,
          toValue: distanceToEdge,
          duration: 250
        }
      )
    ]).start();
  };

  getIcon = () => {
    const { iconName } = this.props;

    return <Icon name={iconName ? "md-" + iconName : !this.state.active ? "md-add" : "md-close"} size={22} color={"#FFF"} />;
  };

  handlePressItem = (itemName) => {
    const { onPressItem } = this.props;

    if (onPressItem) {
      onPressItem(itemName);
    }

    this.reset();
  };

  reset = () => {
    Animated.spring(this.animation, { toValue: 0 }).start();
    Animated.spring(this.actionsAnimation, { toValue: 0 }).start();
    this.updateState({
      active: false
    }, () => {
      if (this.props.onClose) {
        this.props.onClose();
      }
    });
  };

  animateButton = () => {
    const {
      overrideWithAction,
      actions,
      floatingIcon,
      dismissKeyboardOnPress,
      onPressMain
    } = this.props;
    const { active } = this.state;

    if (dismissKeyboardOnPress) {
      Keyboard.dismiss();
    }

    if (overrideWithAction) {
      this.handlePressItem(actions[0].name);

      return;
    }

    if (onPressMain) {
      onPressMain(!active);
    }

    if (!active) {

      Animated.spring(this.actionsAnimation, { toValue: 1 }).start();

      // only execute it for the background to prevent extra calls
      LayoutAnimation.configureNext({
        duration: 180,
        create: {
          type: LayoutAnimation.Types.easeInEaseOut,
          property: LayoutAnimation.Properties.opacity
        }
      });

      this.updateState({
        active: true
      }, () => {
        if (this.props.onOpen) {
          this.props.onOpen();
        }
      });
    } else {
      this.reset();
    }
  };

  renderMainButton() {
    const {
      color,
      overrideWithAction,
      distanceToEdge
    } = this.props;
    const { active } = this.state;

    const position = Platform.OS === "android" ? "right" : "center";

    const animatedVisibleView = {
      opacity: this.fadeAnimation,
      transform: [{
        rotate: this.visibleAnimation.interpolate({
          inputRange: [0, 1],
          outputRange: ['0deg', '90deg']
        })
      }, {
        scale: this.visibleAnimation.interpolate({
          inputRange: [0, 1],
          outputRange: [1, 0]
        })
      }]
    };

    let animatedViewStyle = {
      transform: [{
        rotate: this.animation.interpolate({
          inputRange: [0, 1],
          outputRange: ['0deg', '45deg']
        })
      }]
    };

    if (overrideWithAction) {
      animatedViewStyle = {};
    }

    const Touchable = getTouchableComponent();
    const propStyles = {
      backgroundColor: !active ? color : "#d81717",
      bottom: this.mainBottomAnimation // I need to imporove this to run on native thread and not on JS thread
    };

    if (['left', 'right'].indexOf(position) > -1) {
      propStyles[position] = distanceToEdge;
    }

    return (
      <Animated.View
        style={[
          styles.buttonContainer,
          styles[`${position}Button`],
          propStyles
        ]}
        accessible={true}
        accessibilityLabel={'Floating Action Button'}
      >
        <Touchable
          {...getRippleProps(color)}
          style={styles.button}
          activeOpacity={0.85}
          onPress={this.animateButton}
        >
          <Animated.View style={[styles.buttonTextContainer, animatedViewStyle]}>
            {this.getIcon()}
          </Animated.View>
        </Touchable>
      </Animated.View>
    );
  }

  renderActions() {
    const {
      actions,
      overrideWithAction,
      distanceToEdge,
      actionsPaddingTopBottom
    } = this.props;

    const position = Platform.OS === "android" ? "right" : "center";
    const { active, keyboardHeight } = this.state;

    if (!actions || actions.length === 0) {
      return undefined;
    }

    if (overrideWithAction) {
      return null;
    }

    const animatedActionsStyle = {
      opacity: this.actionsAnimation.interpolate({
        inputRange: [0, 1],
        outputRange: [0, 1]
      })
    };

    const actionsStyles = [
      styles.actions,
      styles[`${position}Actions`],
      animatedActionsStyle,
      {
        bottom: this.actionsBottomAnimation
      }
    ];

    if (active) {
      actionsStyles.push(styles[`${position}ActionsVisible`]);
    }

    const sortedActions = actions.sort((a, b) => a.position - b.position);

    return (
      <Animated.View style={actionsStyles} pointerEvents="box-none">
        {
          sortedActions.map((action) => {
            const textColor = action.textColor || action.actionsTextColor;
            const textBackground = action.textBackground || action.actionsTextBackground;

            return (
              <FloatingButtonItem
                paddingTopBottom={actionsPaddingTopBottom}
                distanceToEdge={distanceToEdge}
                key={action.name}
                textColor={textColor}
                textBackground={textBackground}
                {...action}
                position={position}
                active={active}
                onPress={action.onPress}
              />
            );
          })
        }
      </Animated.View>
    );
  }

  renderTappableBackground() {
    const { overlayColor } = this.props;

    // TouchableOpacity don't require a child
    return (
      <TouchableOpacity
        activeOpacity={1}
        style={[styles.overlay, { backgroundColor: overlayColor }]}
        onPress={this.handlePressBackdrop}
      />
    );
  }

  handlePressBackdrop = () => {
    const { onPressBackdrop } = this.props;
    if (onPressBackdrop) {
      onPressBackdrop();
    }
    this.reset();
  }

  updateState = (nextState, callback) => {
    const { onStateChange } = this.props;
    this.setState(nextState, () => {
      if (callback) {
        callback();
      }
      if (onStateChange) {
        onStateChange({
          isActive: this.state.active
        });
      }
    });
  }

  render() {
    const { active } = this.state;
    const { showBackground } = this.props;

    return (
      <Animated.View
        pointerEvents="box-none"
        style={[styles.overlay, { backgroundColor: 'transparent' }]}
      >
        {
          (active && showBackground) &&
          this.renderTappableBackground()
        }
        {
          this.renderActions()
        }
        {
          this.renderMainButton()
        }
      </Animated.View>
    );
  }
}

FloatingButton.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.shape({
    color: PropTypes.string,
    icon: PropTypes.any,
    name: PropTypes.string.isRequired,
    text: PropTypes.string,
    textBackground: PropTypes.string,
    textColor: PropTypes.string,
    component: PropTypes.func
  })),
  color: PropTypes.string,
  distanceToEdge: PropTypes.number,
  mainVerticalDistance: PropTypes.number,
  visible: PropTypes.bool,
  overlayColor: PropTypes.string,
  position: PropTypes.oneOf(['right', 'left', 'center']),
  overrideWithAction: PropTypes.bool, // replace mainAction with first action from actions
  floatingIcon: PropTypes.any,
  showBackground: PropTypes.bool,
  openOnMount: PropTypes.bool,
  actionsPaddingTopBottom: PropTypes.number,
  iconHeight: PropTypes.number,
  iconWidth: PropTypes.number,
  listenKeyboard: PropTypes.bool,
  dismissKeyboardOnPress: PropTypes.bool,
  onPressItem: PropTypes.func,
  onPressMain: PropTypes.func,
  onClose: PropTypes.func,
  onOpen: PropTypes.func,
  onPressBackdrop: PropTypes.func,
  onStateChange: PropTypes.func
};

FloatingButton.defaultProps = {
  dismissKeyboardOnPress: false,
  listenKeyboard: false,
  actionsPaddingTopBottom: 8,
  overrideWithAction: false,
  visible: true,
  color: '#000',
  overlayColor: 'rgba(68, 68, 68, 0.6)',
  position: 'right',
  distanceToEdge: 30,
  openOnMount: false,
  showBackground: true,
  iconHeight: 15,
  iconWidth: 15,
  mainVerticalDistance: 0
};

const styles = StyleSheet.create({
  actions: {
    position: 'absolute',
    bottom: 85,
    zIndex: 10
  },
  rightActions: {
    alignItems: 'flex-end',
    right: -1000 // this magic number will make always disspear the text from screen
  },
  leftActions: {
    alignItems: 'flex-start',
    left: -1000 // this magic number will make always disspear the text from screen
  },
  centerActions: {
    left: -1000
  },
  rightActionsVisible: {
    right: 0
  },
  leftActionsVisible: {
    left: 0
  },
  centerActionsVisible: {
    left: (DEVICE_WIDTH / 2) - 30
  },
  overlay: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    top: 0,
    elevation: 0,
    zIndex: 0
  },
  buttonContainer: {
    overflow: Platform.OS === 'ios' ? 'visible' : 'hidden',
    zIndex: 2,
    width: ACTION_BUTTON_SIZE,
    height: ACTION_BUTTON_SIZE,
    borderRadius: 28,
    alignItems: 'center',
    justifyContent: 'center',
    shadowOpacity: 0.35,
    shadowOffset: {
      width: 0,
      height: 5
    },
    shadowColor: '#000000',
    shadowRadius: 3,
    elevation: 5,
    position: 'absolute'
  },
  button: {
    zIndex: 3,
    width: ACTION_BUTTON_SIZE,
    height: ACTION_BUTTON_SIZE,
    borderRadius: 28,
    alignItems: 'center',
    justifyContent: 'center'
  },
  rightButton: {},
  leftButton: {},
  centerButton: {
    left: (DEVICE_WIDTH / 2) - 28
  },
  buttonTextContainer: {
    borderRadius: 28,
    width: ACTION_BUTTON_SIZE,
    height: ACTION_BUTTON_SIZE,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  }
});