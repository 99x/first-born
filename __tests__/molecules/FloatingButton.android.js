import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { FloatingButton } from "../../src/molecules/floatingButton/FloatingButton";

jest.mock("Platform", () => {
    const Platform = require.requireActual("Platform");
    Platform.OS = "android";
    return Platform;
});

it("renders regular FloatingButton with a single action", () => {
    const tree = renderer.create(<FloatingButton />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders regular FloatingButton with specific icon", () => {
    const tree = renderer.create(<FloatingButton iconName="heart" />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders regular FloatingButton with multiple actions", () => {
    const actions = [
        {
            text: "Accessibility",
            icon: "help",
            name: "bt_accessibility",
            position: 2
        },
        { text: "Location", icon: "pin", name: "bt_room", position: 1 },
        { text: "Video", icon: "videocam", name: "bt_videocam", position: 3 }
    ];

    const tree = renderer.create(<FloatingButton actions={actions} />).toJSON();
    expect(tree).toMatchSnapshot();
});
