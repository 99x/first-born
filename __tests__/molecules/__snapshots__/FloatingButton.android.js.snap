// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders regular FloatingButton with a single action 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": "transparent",
      "bottom": 0,
      "elevation": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 0,
    }
  }
>
  <View
    accessibilityLabel="Floating Action Button"
    accessible={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#486c86",
        "borderRadius": 28,
        "bottom": 30,
        "elevation": 5,
        "height": 56,
        "justifyContent": "center",
        "overflow": "hidden",
        "position": "absolute",
        "right": 30,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.35,
        "shadowRadius": 3,
        "width": 56,
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300,
          },
          Object {
            "alignItems": "center",
            "borderRadius": 28,
            "height": 56,
            "justifyContent": "center",
            "width": 56,
            "zIndex": 3,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "margin": 20,
          }
        }
      >
        TouchableNativeFeedback is not supported on this platform!
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders regular FloatingButton with multiple actions 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": "transparent",
      "bottom": 0,
      "elevation": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 0,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": 94,
        "opacity": 0,
        "position": "absolute",
        "right": -1000,
        "zIndex": 10,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "elevation": 0,
            "flex": 1,
            "flexDirection": "row",
            "marginBottom": 5,
            "paddingBottom": 8,
            "paddingLeft": 0,
            "paddingRight": 38,
            "paddingTop": 8,
            "paddingVertical": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 4,
                "elevation": 5,
                "height": 22,
                "marginTop": 8,
                "paddingHorizontal": 8,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
                "shadowOpacity": 0.35,
                "shadowRadius": 3,
              },
              Object {
                "marginRight": 14,
              },
              Object {
                "backgroundColor": "#FFF",
                "elevation": 5,
                "shadowOffset": Object {
                  "height": 5,
                },
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 16,
                  "fontWeight": "normal",
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(33, 33, 33, 0.87)",
                  "fontSize": 14,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Location
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 20,
                "elevation": 5,
                "height": 40,
                "justifyContent": "center",
                "padding": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
                "shadowOpacity": 0.35,
                "shadowRadius": 3,
                "width": 40,
              },
              Object {
                "backgroundColor": "#486c86",
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#FFF",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "elevation": 0,
            "flex": 1,
            "flexDirection": "row",
            "marginBottom": 5,
            "paddingBottom": 8,
            "paddingLeft": 0,
            "paddingRight": 38,
            "paddingTop": 8,
            "paddingVertical": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 4,
                "elevation": 5,
                "height": 22,
                "marginTop": 8,
                "paddingHorizontal": 8,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
                "shadowOpacity": 0.35,
                "shadowRadius": 3,
              },
              Object {
                "marginRight": 14,
              },
              Object {
                "backgroundColor": "#FFF",
                "elevation": 5,
                "shadowOffset": Object {
                  "height": 5,
                },
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 16,
                  "fontWeight": "normal",
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(33, 33, 33, 0.87)",
                  "fontSize": 14,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Accessibility
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 20,
                "elevation": 5,
                "height": 40,
                "justifyContent": "center",
                "padding": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
                "shadowOpacity": 0.35,
                "shadowRadius": 3,
                "width": 40,
              },
              Object {
                "backgroundColor": "#486c86",
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#FFF",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "elevation": 0,
            "flex": 1,
            "flexDirection": "row",
            "marginBottom": 5,
            "paddingBottom": 8,
            "paddingLeft": 0,
            "paddingRight": 38,
            "paddingTop": 8,
            "paddingVertical": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 4,
                "elevation": 5,
                "height": 22,
                "marginTop": 8,
                "paddingHorizontal": 8,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
                "shadowOpacity": 0.35,
                "shadowRadius": 3,
              },
              Object {
                "marginRight": 14,
              },
              Object {
                "backgroundColor": "#FFF",
                "elevation": 5,
                "shadowOffset": Object {
                  "height": 5,
                },
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 16,
                  "fontWeight": "normal",
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(33, 33, 33, 0.87)",
                  "fontSize": 14,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Video
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 20,
                "elevation": 5,
                "height": 40,
                "justifyContent": "center",
                "padding": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
                "shadowOpacity": 0.35,
                "shadowRadius": 3,
                "width": 40,
              },
              Object {
                "backgroundColor": "#486c86",
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#FFF",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityLabel="Floating Action Button"
    accessible={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#486c86",
        "borderRadius": 28,
        "bottom": 30,
        "elevation": 5,
        "height": 56,
        "justifyContent": "center",
        "overflow": "hidden",
        "position": "absolute",
        "right": 30,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.35,
        "shadowRadius": 3,
        "width": 56,
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300,
          },
          Object {
            "alignItems": "center",
            "borderRadius": 28,
            "height": 56,
            "justifyContent": "center",
            "width": 56,
            "zIndex": 3,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "margin": 20,
          }
        }
      >
        TouchableNativeFeedback is not supported on this platform!
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders regular FloatingButton with specific icon 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": "transparent",
      "bottom": 0,
      "elevation": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 0,
    }
  }
>
  <View
    accessibilityLabel="Floating Action Button"
    accessible={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#486c86",
        "borderRadius": 28,
        "bottom": 30,
        "elevation": 5,
        "height": 56,
        "justifyContent": "center",
        "overflow": "hidden",
        "position": "absolute",
        "right": 30,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.35,
        "shadowRadius": 3,
        "width": 56,
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300,
          },
          Object {
            "alignItems": "center",
            "borderRadius": 28,
            "height": 56,
            "justifyContent": "center",
            "width": 56,
            "zIndex": 3,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "margin": 20,
          }
        }
      >
        TouchableNativeFeedback is not supported on this platform!
      </Text>
    </View>
  </View>
</View>
`;
